cmd/data-usage-cache.go

```diff
@@ -46,14 +46,15 @@ type sizeHistogram [dataUsageBucketLen]uint64
 //msgp:tuple dataUsageEntry
 type dataUsageEntry struct {
 	// These fields do no include any children.
-	Size                   int64
-	ReplicatedSize         uint64
-	ReplicationPendingSize uint64
-	ReplicationFailedSize  uint64
-	ReplicaSize            uint64
-	Objects                uint64
-	ObjSizes               sizeHistogram
-	Children               dataUsageHashMap
+	Size                    int64
+	ReplicatedSize          uint64
+	ReplicationPendingSize  uint64
+	ReplicationFailedSize   uint64
+	ReplicaSize             uint64
+	Objects                 uint64
+	ObjSizes                sizeHistogram
+	Children                dataUsageHashMap
+	ReplicationPendingCount uint64
```

Comment:
Do not add more entries here - this will cause dataUsageEntry to bloat and cause more memory usage. We should avoid it and keep it outside if not needed inside usage.

Reply:
With this change, you also need to bump up the dataUsageEntry version such that it's updated otherwise it won't be read and it will clear off the old values.. look at

```go
// deserialize the supplied byte slice into the cache.
func (d *dataUsageCache) deserialize(r io.Reader) error {
        var b [1]byte
        n, _ := r.Read(b[:])
        if n != 1 {
                return io.ErrUnexpectedEOF
        }
```

Reply:
We should also put this in a struct IMO such as 

```
type ReplicationStats struct {
     PendingSize
     FailedSize
     FailedCount
     PendingCount
     MissedThresholdSize
     AfterThresholdSize
     MissedThresholdCount
     AfterThresholdCount
}
```

We need all of this information and it looks like dataUsageEntry is not the place to calculate this - we have to come up with a better mechanism.